Refresh FinOps hub upgrade procedure for multi-version FOCUS adoption (#2123)
microsoft/finops-toolkit: src/templates/finops-hub/docs/README.md

- Each versioned function unions data from versioned tables in the Ingestion database and transforms it to that FOCUS version for back compat.
- Consumers should use the unversioned function for the latest and the versioned functions for back compat.

To add a new FOCUS version:

0. Confirm dependencies

>    1. Ensure the FOCUS specification has been ratified for the target version. Support for working draft FOCUS versions should not be shipped before official ratification to prevent customer churn sinc FOCUS working drafts may change without notice.

Reviewer: Typo: `sinc` -> `since`.

Author: 🤖 [AI][Claude Code] ✅ **Implemented**

Already fixed — `sinc` → `since` in README.md step 0.1 (shipped in 463c2735). Verified 0 remaining `sinc` hits.

>    2. Check whether Microsoft Cost Management has shipped a matching FOCUS export dataset version. The hub depends on a `focuscost_X.Y.json` schema mapping file in [Microsoft.CostManagement/Exports/schemas](../modules/Microsoft.CostManagement/Exports/schemas/) when ingesting from Cost Management. If the export is not yet available, the hub schema can still ship as GA &mdash; the hub will ingest the new format as soon as the export ships. Note the gap in the changelog so adopters know what additional setup will be required.

Author: ```suggestion
0. Confirm dependencies
   1. Ensure the FOCUS specification has been ratified for the target version. Support for working draft FOCUS versions should not be shipped before official ratification to prevent customer churn sinc FOCUS working drafts may change without notice.
   2. Check whether Microsoft Cost Management has shipped a matching FOCUS export dataset version. The hub depends on a `focuscost_X.Y.json` schema mapping file in [Microsoft.CostManagement/Exports/schemas](../modules/Microsoft.CostManagement/Exports/schemas/) when ingesting from Cost Management. If the export is not yet available, the hub schema can still ship as GA. Note the gap in the changelog so adopters know what additional setup will be required.
```

>    2. Check whether Microsoft Cost Management has shipped a matching FOCUS export dataset version. The hub depends on a `focuscost_X.Y.json` schema mapping file in [Microsoft.CostManagement/Exports/schemas](../modules/Microsoft.CostManagement/Exports/schemas/) when ingesting from Cost Management. If the export is not yet available, the hub schema can still ship as GA. Note the gap in the changelog so adopters know what additional setup will be required.

Reviewer: The PR summary says step 0 should instruct shipping hub support as **preview** when the Cost Management export is not yet available, but this line says ship as **GA** (and data-model.md has no preview-convention text). Please reconcile the description with what shipped here. The body also lists "steps 5-8" but only 5-7 landed.

Author: The outdated summary needs to be updated

Author: 🤖 [AI][Claude Code] ✅ **Reconciled**

PR body updated to match what shipped: step 0.2 now states the hub schema can ship as **GA** when the Cost Management export is unavailable (no "preview" claim), and the step count is corrected to **steps 5-7**. The `-preview` convention is now documented in both README.md step 5.1 and data-model.md. Also corrected the stale `ms.date` reference in the PR body to 07/25/2026 (the CI bot bumped it).

   2. Update the KQL reports to use the new versioned functions
   3. Update the ADX dashboard to use the new versioned functions
   4. Update the FOCUS queries in the best practices library to use the new versioned functions
5. Update open-data metadata

>    1. Create a `FocusCost_<version>.json` file into [src/open-data/dataset-metadata](../../../open-data/dataset-metadata/) for each Cost Management cost export version.

Reviewer: Nit: "file **into** [folder]" -> "file **in** [folder]". Same on the next line (5.2).

Author: 🤖 [AI][Claude Code] ✅ **Implemented**

Already fixed — "file into [folder]" → "file in [folder]" in both README.md steps 5.1 and 5.2 (shipped in 463c2735). Verified 0 remaining `file into` hits.

   2. Update the KQL reports to use the new versioned functions
   3. Update the ADX dashboard to use the new versioned functions
   4. Update the FOCUS queries in the best practices library to use the new versioned functions
5. Update open-data metadata
   1. Drop a new `FocusCost_<version>.json` file into [src/open-data/dataset-metadata](../../../open-data/dataset-metadata/).

>    2. Mirror the schema details (columns, types, descriptions) from the matching Cost Management export schema so downstream consumers see consistent metadata.

Author: ```suggestion
   1. Create a `FocusCost_<version>.json` file into [src/open-data/dataset-metadata](../../../open-data/dataset-metadata/) for each Cost Management cost export version.
   2. Create a `FinOpsHubs_<dataset>_<schema-version>.json` file into [src/open-data/dataset-metadata](../../../open-data/dataset-metadata/) for each FinOps hub managed dataset schema version.
   3. Mirror the schema details (columns, types, descriptions) from the matching Cost Management export or FinOps hubs schema so downstream consumers see consistent metadata.
```

7. Update changelog
   1. Add an entry under the next version in [docs-mslearn/toolkit/changelog.md](../../../../docs-mslearn/toolkit/changelog.md) describing the new FOCUS version support and any preview status.

### Handling multiple FOCUS versions in one cycle

Occasionally, the toolkit needs to support two FOCUS versions in a single release &ndash; for example, a newly ratified version alongside a working draft of the next version. When that happens:

- The older version follows the standard `_v1_X` naming and ships as generally available (GA).
- The newer version uses the next `_v1_Y` suffix and is labeled **preview** in user-facing documentation, including [data-model.md](../../../../docs-mslearn/toolkit/hubs/data-model.md) and [changelog.md](../../../../docs-mslearn/toolkit/changelog.md).
- Preview schemas may change without notice between releases. Treat them as opt-in for early adopters only.
- The unversioned functions (`Costs()`, `Prices()`, etc.) alias to the latest **GA** schema, not the preview. The aliases promote to the newer version only after it transitions from preview to GA.

> This guarantees backwards compatibility for production consumers while still enabling early validation of the next FOCUS version.

Author: ```suggestion
```
